pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -60,6 +64,19 @@ func (pl *TaintToleration) EventsToRegister() []framework.ClusterEvent {
 	}
 }
 
+// PreFilter builds and writes cycle state used by Filter.
+func (pl *TaintToleration) PreFilter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
+	if len(pod.Spec.Tolerations) != 0 {
+		return &framework.PreFilterResult{NodeNames: sets.NewString(pod.Spec.NodeName)}, nil
+	}
+	return nil, framework.NewStatus(framework.Skip, ErrReasonNoToleration)
```

Comment:
If we can really return Skip when Pod doesn't have any `Tolerations`?
No `Tolerations` doesn't mean TaintToleration won't do anything in Fitler because it still needs to filter out Nodes with Taint.

Reply:
So, I guess probably TaintToleration cannot return Skip in PreFilter. For `TaintToleration` plugin, it needs to know if there are any Nodes with un-tolerated taints. And I feel it doesn't make sense to do that in PreFilter because it just brings what it does in FIlter to PreFilter. wdyt

Reply:
I think so, too. PreFilter can't confirm nodes,  so TaintToleration PreFilter can't improve the scheduling.

I close this PR. thanks.